### PR TITLE
Debug CI Issue

### PR DESCRIPTION
**Pull request by WillAyd**

### CODE REVIEW

**Comment by simonjayhawkins** [comment]
lint error ```##[error]./pandas/tests/extension/base/dtype.py:1:1:F401:'warnings' imported but unused```

**Comment by TomAugspurger** [comment]
@WillAyd I have  some time over the next couple hours to debug this. I'll plan to push to your branch if that's OK.

**Comment by TomAugspurger** [comment]
Ends up being pretty simple. With older numpys

```python
In [24]: pd.DataFrame({"A": pd.array([], dtype="Int64"), "B": pd.array([], dtype="int64")}).dtypes == 'Int64'
Out[24]:
A    True
B    True
dtype: bool

In [25]: np.dtype("int64") == "Int64"
/Users/taugspurger/.virtualenvs/dask-dev/bin/ipython:1: DeprecationWarning: Numeric-style type codes are deprecated and will result in an error in the future.
  #!/Users/taugspurger/Envs/dask-dev/bin/python
Out[25]: True
```

With NumPy master

```python
In [1]: import pandas as pd
i
In [2]: import numpy as np

In [3]: pd.DataFrame({"A": pd.array([], dtype="Int64"), "B": pd.array([], dtype="int64")}).dtypes == 'Int64'
Out[3]:
A     True
B    False
dtype: bool

In [4]: np.dtype("int64") == "Int64"
---------------------------------------------------------------------------
TypeError                                 Traceback (most recent call last)
<ipython-input-4-49749ad32aa3> in <module>
----> 1 np.dtype("int64") == "Int64"

TypeError: data type 'Int64' not understood
```

Will push a fix up in a bit.

File: pandas/compat/numpy/__init__.py [hunk 1]
```diff
@@ -11,6 +11,8 @@
 _np_version_under1p16 = _nlv < LooseVersion("1.16")
 _np_version_under1p17 = _nlv < LooseVersion("1.17")
 _np_version_under1p18 = _nlv < LooseVersion("1.18")
+_np_version_under1p19 = _nlv < LooseVersion("1.19")
+_np_version_under1p20 = _nlv < LooseVersion("1.20")
```
Comment (TomAugspurger) [inline pandas/compat/numpy/__init__.py:15]: I think this is targeted for NumPy 1.20, but verifying in https://github.com/numpy/numpy/pull/16554#issuecomment-643232285.

File: pandas/tests/extension/base/dtype.py [hunk 2]
```diff
@@ -70,13 +72,13 @@ def test_check_dtype(self, data):
...
         # so can't distinguish
-        if dtype.name == "Int64":
+        if dtype.name == "Int64" and _np_version_under1p20:
+            # TODO(numpy-1.20): This if block and the warnings filter can be removed
+            # once we require numpy>=1.20
             expected = pd.Series([True, True, False, True], index=list("ABCD"))
         else:
             expected = pd.Series([True, True, False, False], index=list("ABCD"))
 
-        # FIXME: This should probably be *fixed* not ignored.
```
Comment (TomAugspurger) [inline pandas/tests/extension/base/dtype.py:78]: I'm not sure that there's anything we can actually do here. Open to suggestions, but the fact is that NumPy controls the behavior of `np.dtype.__eq__`. So I'm comfortable just leaving this for NumPy<1.20.

**Comment by TomAugspurger** [comment]
cc @jreback if you have a chance to look. Planning to merge on green to get CI running again though.

**Comment by jreback** [comment]
lgtm merge when ready

**Comment by jreback** [comment]
did we merge something to fix CI?

**Comment by TomAugspurger** [comment]
Sorry, thought this was in. Taking another look.

**Comment by TomAugspurger** [comment]
Ah, I think this issue is that the numpydev build is NumPy 1.20.dev0.stuff, but it's not new enough to have the changes in the travis-CI job (built from numpy source).

**Comment by TomAugspurger** [comment]
Right now https://anaconda.org/scipy-wheels-nightly/numpy/files doesn't have a wheel with the new behavior the travis build (building numpy from source) is seeing.

So it's ugly, but we have to make the check dependent on the behavior of `np.dtype('int64') == "Int64"` until we can get a wheel with the new behavior.  Then we can just check the version number.

**Comment by WillAyd** [comment]
Maybe just skip the test until the wheel is available?

**Comment by TomAugspurger** [comment]
Slight preference for this, just in case we forget to update or it takes a little while for new wheels to show up.

**Comment by TomAugspurger** [comment]
No idea why this started failing now

```
        with tm.assert_produces_warning(UserWarning):
            axes = _check_plot_works(
>               scatter_matrix, filterwarnings="always", frame=df, range_padding=0.1
            )

pandas\tests\plotting\test_misc.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _

self = <contextlib._GeneratorContextManager object at 0x000001A3127D1F48>
type = None, value = None, traceback = None

    def __exit__(self, type, value, traceback):
        if type is None:
            try:
>               next(self.gen)
E               AssertionError: Caused unexpected warning(s): [('DeprecationWarning', DeprecationWarning("Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working"), 'C:\\Miniconda\\envs\\pandas-dev\\lib\\site-packages\\matplotlib\\cbook\\__init__.py', 2349), ('DeprecationWarning', DeprecationWarning("Using or importing the ABCs from 'collections' instead of from 'collections.abc' is deprecated since Python 3.3,and in 3.9 it will stop working"), 'C:\\Miniconda\\envs\\pandas-dev\\lib\\site-packages\\matplotlib\\cbook\\__init__.py', 2366)]
```

**Comment by WillAyd** [comment]
Thanks @TomAugspurger - pretty tricky stuff here

**Comment by simonjayhawkins** [comment]
@MeeseeksDev backport to 1.0.x

**Pull request merged**